src/main/kotlin/com/criptext/mail/scenes/restorebackup/RestoreBackupScene.kt

```diff
@@ -106,24 +122,41 @@ interface RestoreBackupScene{
             }
         }
 
-        override fun showBackupFoundLayout(isEncrypted: Boolean) {
-            cloudIcon.setImageResource(R.drawable.restore_cloud)
+        override fun showBackupFoundLayout(isEncrypted: Boolean, isLocal: Boolean) {
+            when {
+                isLocal && isEncrypted -> {
+                    cloudIcon.setImageResource(R.drawable.img_restoremail)
+                    textViewTitle.text = context.getLocalizedUIMessage(UIMessage(R.string.restore_backup_local_title))
+                    passwordInput.visibility = View.VISIBLE
+                }
+                isLocal && !isEncrypted -> {
+                    cloudIcon.setImageResource(R.drawable.img_restoremail)
+                    textViewTitle.text = context.getLocalizedUIMessage(UIMessage(R.string.restore_backup_local_title))
+                    passwordInput.visibility = View.GONE
+                }
+                !isLocal && isEncrypted -> {
```

Comment:
Encrypted from Google Drive? is that possible?

Reply:
Yes, you can export it manually encrypted, and save it to drive, since the SAF allows you to save it to drive, or even dropbox or whichever Service Provider that implements the android access framework.